lib/administration/user/src/user_administration.cpp

```diff
@@ -85,18 +85,29 @@ namespace irods::experimental::administration::NAMESPACE_IMPL
 
     auto add_group(RxComm& _comm, const group& _group) -> void
     {
-        const auto zone = get_local_zone(_comm);
+        const auto zone{get_local_zone(_comm)};
 
-        GeneralAdminInput input{};
-        input.arg0 = "add";
-        input.arg1 = "user";
-        input.arg2 = _group.name.data();
-        input.arg3 = "rodsgroup";
-        input.arg4 = zone.data();
+        const auto current_user{user{_comm.clientUser.userName, _comm.clientUser.rodsZone}};
+        const auto user_type{type(_comm, current_user)};
 
-        if (const auto ec = rxGeneralAdmin(&_comm, &input); ec != 0) {
-            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
-            THROW(ec, fmt::format("Failed to add group [{}].", _group.name));
+        if (user_type == irods::experimental::administration::user_type::groupadmin) {
+            userAdminInp_t input{.arg0 = "mkgroup",
+                                 .arg1 = const_cast<char*>(_group.name.data()),
+                                 .arg2 = "rodsgroup",
+                                 .arg3 = const_cast<char*>(zone.data())};
```

Comment:
Are there other members in that struct?
Are they guaranteed to be initialized?
Are we required by the C++ standard to initialize all members?

Reply:
> Are there other members in that struct?

Yes, the struct goes from arg0 to arg9, much like `GeneralAdminInput`.

> Are they guaranteed to be initialized?

Relevant Page: https://en.cppreference.com/w/cpp/language/aggregate_initialization

Partial quote:

> ...
> For a non-union aggregate, each element that is not an explicitly initialized element is initialized as follows:
> - If the element has a default member initializer, the element is initialized from that initializer. (since C++11)
> - Otherwise, if the element is not a reference, the element is copy-initialized from an empty initializer list.
> - Otherwise, the program is ill-formed. 
>
> ...

So in this case, since the struct contains 10 `char*`, the ones not explicitly defined behave as if initialized as follows: `char* arg# {};`.

That would mean it does zero initialization: https://en.cppreference.com/w/cpp/language/zero_initialization.

This would then would apply the following rule, as a pointer is a scalar type:
> ...
> The effects of zero-initialization are:
> - If T is a scalar type, the object is initialized to the value obtained by explicitly converting the integer literal ​0​ (zero) to T.
>
> ...

Very good question, made me learn a good bit about how this works too. 👍 

> Are we required by the C++ standard to initialize all members?

I think the question before answered this one, but not initializing the pointers might mean we try to access something that doesn't exist, right?

Anywho, I think by doing `userAdminInp_t input;`, then initializing the elements thereafter would avoid initializing all elements.

Reply:
Good. Check the compiler implementation status for clang 13.0.1 to be sure it fully implements the feature.

> Anywho, I think by doing `userAdminInp_t input;`, then initializing the elements thereafter would avoid initializing all elements.

We'd make it clear its members by adding a pair of curly braces to it (e.g. `userAdminInp_t input{}`).

Reply:
Supported since clang 10, it's a subset(?) of aggregate initialization called Designated initializers. https://en.cppreference.com/w/cpp/compiler_support/20

Reply:
And GCC8 it looks like

Reply:
Good. I don't have a strong preference on this. Seems fine to me.

Reply:
Think this is fine for new code, closing now.